service/worker/archiver/activities.go

```diff
@@ -149,6 +150,13 @@ func archiveVisibilityActivity(ctx context.Context, request ArchiveRequest) (err
 		logger.Error(carchiver.ArchiveNonRetryableErrorMsg, tag.ArchivalArchiveFailReason("failed to get visibility archiver"), tag.Error(err))
 		return errArchiveVisibilityNonRetryable
 	}
+
+	// It is safe to pass nil to typeMap here because search attributes type must be embedded by caller.
+	searchAttributes, err := searchattribute.Stringify(request.SearchAttributes, nil)
+	if err != nil {
+		logger.Error("Unable to stringify search attributes.", tag.Error(err))
```

Comment:
return err

Reply:
`Stringify` do partial processing and if some of search attributes can't be converted it continues to the next one. This is ok for CLI, for example, but not for archive. Fixed.